app/views/questionnaire.py

```diff
@@ -191,6 +193,20 @@ def get_block_handler(
     answer_store = questionnaire_store.answer_store
 
     current_location = Location(block_id, list_name, list_item_id)
+    try:
+        current_section_id = schema.section(block_id=block_id)['id']
+    except (AttributeError, TypeError):
+        current_section_id = None
+
+    if current_location in questionnaire_store.completed.blocks:
+        questionnaire_store.completed.blocks.remove(current_location)
+        questionnaire_store.add_or_update()
+
+    if current_section_id in questionnaire_store.completed.sections and schema.block(
+        current_location.block_id
+    )['type'] not in ['SectionSummary', 'Summary']:
+        questionnaire_store.completed.sections.remove(current_section_id)
```

Comment:
Adding is done via the `add_completed_section` method on questionnaire store updater, remove should be consistent.